### PR TITLE
Feature/spliced probability dist

### PR DESCRIPTION
**Pull request by apmccartney**
@jlconlin Let's try our first github pull review.

This pull request is composed of
- A new class called `SplicedProbabilityDistribution` which provides continuous interpolation between two probability scheme, given their conditions and a queried condition. The intent of this class is to provide the machinery for the `givenCondition` method added to the `BivariateProbabilityDistribution` class (see below). The spliced probability distribution only stores references to the boundary probability distributions and hence its validity is limited to the scope of those objects.
- The `BivariateProbabilityDistribution` has been extended to provide a probability distribution object for an arbitrary queried condition value. This method has been labeled `givenCondition` and returns a `unique_ptr<math::API::ProbabilityDistribution>` which point to a spliced probability distribution. 
- I cleaned up the `BivariateProbabilityDistribution` tests (the same way that was done to the `EnergySpectrum` in nuclear data).
- A README.md has been added with all the relevant badges.
- Minor revisions to logging messages to provide additional information or clarify cause of error.
- Minor revisions to documentation, cleaning up a handful of typos and providing clarification in some places.

### CODE REVIEW

File: implementation/BivariateProbabilityDistribution/src/conditionalCumulativeProbability.hpp.in [hunk 1]
```diff
@@ -4,12 +4,16 @@
...
+ * \f$\int_{-\infty}^{\mathrm{event}}P(mathrm{event'} | mathrm{condition])~d mathrm{event'}\f$
  *
- * @returns the probability of @p event given the @p condition
+ * @returns 
+ * The cumulative probability function given the @p condition evalauted at @p event 
  *
- * @details It is the responsibility of the whoever calls this method to verify
- * that the event is valid.
+ * @details
+ * It is the responsibility of the method caller to validate that the event value
```
Comment (jlconlin) [inline implementation/BivariateProbabilityDistribution/src/conditionalCumulativeProbability.hpp.in]: Incomplete thought.

File: implementation/BivariateProbabilityDistribution/src/conditionalProbabilityDensity.hpp.in [hunk 2]
```diff
@@ -4,12 +4,14 @@
...
+ * @brief The conditional probability density ie \f$P(@p event | @p condition)\f$
  *
- * @returns the probability of @p event given the @p condition
+ * @returns 
+ * The probability density function given the @p condition evalauted at @p event 
  *
- * @details It is the responsibility of the whoever calls this method to verify
- * that the event is valid.
+ * @details 
+ * It is the responsibility of the method caller to validate that the event value
```
Comment (jlconlin) [inline implementation/BivariateProbabilityDistribution/src/conditionalProbabilityDensity.hpp.in]: incomplete sentence.

File: implementation/SplicedProbabilityDistribution/src/SplicedProbabilityDistribution.hpp.in [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#ifndef MATH_IMPLEMENTATION_SPLICEDPROBABILITYDISTRIBUTION_SRC_SPLICEDPROBABILITYDISTRIBUTION_HPP
+#define MATH_IMPLEMENTATION_SPLICEDPROBABILITYDISTRIBUTION_SRC_SPLICEDPROBABILITYDISTRIBUTION_HPP
+
+#include <algorithm>
+
+#include "math/implementation/SplicedProbabilityDistribution.hpp"
+
```
Comment (jlconlin) [inline implementation/SplicedProbabilityDistribution/src/SplicedProbabilityDistribution.hpp.in]: Documentation?

**Comment by jlconlin** [comment]
Very nice. Just two comments about the documentation that should be addressed. They are very minor.

**Comment by apmccartney** [comment]
These documentation was cleaned up in the follow up pull

**Pull request merged**